xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -428,6 +437,30 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
       // Offset this line's starting position
       startX -= w;
     }
+    else if (alignment & XBFONT_RIGHT && alignment & XBFONT_TRUNCATED)
```

Comment:
I think there is a problem with right alignment and no truncated flag.
If alignment parameter is initially RIGHT | TRUNCATED, then TRUNCATED is removed @line 407 when the text fits > this block is skipped, OK
If alignment parameter is RIGHT | TRUNCATED, and the text doesn't fit > handled by this block, OK
If alignment parameter is RIGHT without TRUNCATED and the text doesn't fit > not handled, or I missed it

Without TRUNCATED I'm not sure if you should stop at the last character that will not be clipped at all, or the next one that will be partially clipped. Should be the same as what is done by current code for left alignment and no TRUNCATED.